site/source/tutorials/1010_containers_and_injecting_state/index.md

```diff
@@ -0,0 +1,177 @@
+---
+layout: false
+title: State management
+overview: In this tutorial, you will learn what a Containers and Injectors are and how to use them to manage external state and inject this into the parts of the widget tree.
+---
+
+{% section 'first' %}
+
+# State management
+
+## Overview
+
+Almost all non-trivial applications have multiple states that they can exist in. While Dojo 2's widgets can manage application state, they can quickly become overburdened as they manage their own visual representations, listen for interactions from the user, manage their children, and keep track of state information. Additionally, using widgets to pass state through an application often forces the widgets to be aware of state information for the sole purpose of passing that data down to their children. To allow widgets to remain focused on their primary roles of providing a visual representation of themselves and listening for user interactions, Dojo 2 provides `Containers` and `Injectors` that are designed to manage an application's state and provide that information only to the widgets that require it.
+
+In this tutorial, we will start with an application that is using widgets to manage its state. We will then extract all of the state-related code out of the widgets and inject it into the widgets that need it. You can [download](../assets/1010_form_widgets-initial.zip) the demo project to get started.
```

Comment:
"inject it into the widgets that need it" -> "inject state only into the widgets as needed"